core/trino-main/src/main/java/io/trino/sql/planner/RelationPlanner.java

```diff
@@ -441,10 +441,11 @@ private RelationPlan addColumnMasks(Table table, RelationPlan plan)
         for (int i = 0; i < plan.getDescriptor().getAllFieldCount(); i++) {
             Field field = plan.getDescriptor().getFieldByIndex(i);
 
-            io.trino.sql.tree.Expression mask = columnMasks.get(field.getName().orElseThrow());
+            Optional<io.trino.sql.tree.Expression> columnMask = Analysis.resolveColumnMask(table.getName(), field.getName().orElseThrow(), columnMasks);
```

Comment:
I don't understand this. You pass multiple columns (field expressions) and you get a single mask. I think you should pass only a single column here.

Reply:
This shouldn't be doing any resolution. It has the `Field`, so all it needs to do is look up the corresponding expression for that field, which should be stored in the analysis object.

So, in a nutshell:
* The analyzer resolves the schema associated with the underlying table (modeled by RelationType)
* The analyzer resolves the masks by column name and analyzes them (type checks, etc). It associates them with the corresponding Field from RelationType.
* The planner injects the necessary projections for each Field by looking them up in the data structure produces by the analyzer (Analysis)

Reply:
Thanks! I can see there are already `Map`s where `Field` is the key - I wasn't sure I can use it this way, but now the `columnMasks` collection simply changes from `Map<String, Expression>` to `Map<Field, Expression>` and it simplifies things a bit.